interface/super/layout_service_codes.php

```diff
@@ -0,0 +1,224 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Rod Roark <rod@sunsetsystems.com>
+ * @link    http://www.open-emr.org
+ */
+
+// Disable magic quotes and fake register globals.
+$sanitize_all_escapes = true;
+$fake_register_globals = false;
+
+require_once('../globals.php');
+require_once($GLOBALS['srcdir'] . '/acl.inc');
+require_once($GLOBALS['srcdir'] . '/htmlspecialchars.inc.php');
+require_once($GLOBALS['srcdir'] . '/formdata.inc.php');
+require_once($GLOBALS['fileroot'] . '/custom/code_types.inc.php');
+
+if (!acl_check('admin', 'super')) die(xlt('Not authorized'));
+
+$form_dryrun = !empty($_POST['form_dryrun']);
+
+function applyCode($layoutid, $codetype, $code, $description) {
+  global $thecodes;
+  if (!isset($thecodes[$layoutid])) $thecodes[$layoutid] = array();
+  $thecodes[$layoutid]["$codetype:$code"] = $description;
+}
+
+?>
+<html>
+
+<head>
+<title><?php echo xlt('Install Layout Service Codes'); ?></title>
+<link rel="stylesheet" href='<?php echo $css_header ?>' type='text/css'>
+
+<style type="text/css">
+ .dehead { color:#000000; font-family:sans-serif; font-size:10pt; font-weight:bold }
+ .detail { color:#000000; font-family:sans-serif; font-size:10pt; font-weight:normal }
+</style>
+
+</head>
+
+<body class="body_top">
+
+<?php
+// Handle uploads.
+if (!empty($_POST['bn_upload'])) {
+  $thecodes = array();
+  $tmp_name = $_FILES['form_file']['tmp_name'];
+
+  if (is_uploaded_file($tmp_name) && $_FILES['form_file']['size']) {
+    $fhcsv = fopen($tmp_name, 'r');
+    if (empty($fhcsv)) die(xlt('Cannot open') . text(" '$tmp_name'"));
+
+    // Columns are:
+    // 0 - Layout ID, e.g. LBFVIA
+    // 1 - Code type, e.g. IPPF2
+    // 2 - Code
+    //
+    while (!feof($fhcsv)) {
+      $codecount = 0;
+      $acsv = fgetcsv($fhcsv, 1024);
+      if (count($acsv) < 3) continue;
+      $layoutid = trim($acsv[0]);
+      $codetype = trim($acsv[1]);
+      $code     = trim($acsv[2]);
+      if (empty($layoutid) || empty($codetype) || empty($code)) continue;
+      // If this is already a Fee Sheet code, add it.
+      if (empty($code_types[$codetype]['nofs'])) {
+        applyCode($layoutid, $codetype, $code, xl('Direct'));
+        ++$codecount;
+      }
+      // Add all Fee Sheet codes that relate to this code.
+      foreach ($code_types as $ct_key => $ct_arr) {
+        if (!$ct_arr['active'] || $ct_arr['nofs']) continue;
+        $tmp = "$codetype:$code";
+        $relres = sqlStatement("SELECT code, code_text FROM codes WHERE code_type = ? AND " .
+          "(related_code LIKE ? OR related_code LIKE ? OR related_code LIKE ? OR related_code LIKE ?) " .
+          "AND active = 1 ORDER BY code",
```

Comment:
just making sure this does not need to incorporate external codes

Reply:
True and at present this is specific to IPPF and is not in the menus.